use cortex-m-generic.ld instead of libopencm3_stm32f4.ld

this seems like an upstream change in libopencm3, but this helped me compile your project.

there's no libopencm3_stm32f4.ld anywhere

```
diff --git a/src/bootloader/lctech-f103.ld b/src/bootloader/lctech-f103.ld
index 56dde87..a6e8e9e 100644
--- a/src/bootloader/lctech-f103.ld
+++ b/src/bootloader/lctech-f103.ld
@@ -28,5 +28,4 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f1.ld
-
+INCLUDE cortex-m-generic.ld
diff --git a/src/lctech-f103.ld b/src/lctech-f103.ld
index fc5714e..00b739d 100644
--- a/src/lctech-f103.ld
+++ b/src/lctech-f103.ld
@@ -28,5 +28,4 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f1.ld
-
+INCLUDE cortex-m-generic.ld
diff --git a/src/stm32f4-discovery.ld b/src/stm32f4-discovery.ld
index 927c786..c72a9bf 100644
--- a/src/stm32f4-discovery.ld
+++ b/src/stm32f4-discovery.ld
@@ -28,5 +28,5 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f4.ld
+INCLUDE cortex-m-generic.ld
 
```

I am trying to compile the project with the following Dockerfile:

```
FROM ubuntu:18.04
LABEL Description="I2C-star for STM32 Bluepill board"          

RUN DEBIAN_FRONTEND=noninteractive apt-get update -y -q && apt-get install -y -q sudo make python gcc-arm-none-eabi git-core libnewlib-arm-none-eabi

ENV user i2cstar
RUN useradd -d /home/$user -m -s /bin/bash $user
RUN echo "$user ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/$user
RUN chmod 0440 /etc/sudoers.d/$user

USER $user
WORKDIR /home/$user
RUN mkdir -pv code
COPY . ./code/
RUN sudo chown $user.$user -R /home/$user/code
WORKDIR /home/$user/code/
RUN git submodule update --init --recursive
RUN make
```

Then I end up with a similar error:

```
  CC      sync.c
  CC      dwt.c
  AR      libopencm3_stm32f4.a
  BUILD   src/bootloader
  CC      i2c_ctx.c
  CC      bitops.c
  CC      fibre.c
  CC      fibre_default.c
  CC      list.c
  CC      messageq.c
  CC      regdump.c
  CC      ringbuf.c
  CC      time_libopencm3.c
  CC      util.c
  CC      usbdfu.c
  LD      usbdfu.elf
/usr/lib/gcc/arm-none-eabi/6.3.1/../../../arm-none-eabi/bin/ld: cannot open linker script file libopencm3_stm32f1.ld: No such file or directory
collect2: error: ld returned 1 exit status
../../mk/libopencm3.rules.mk:158: recipe for target 'usbdfu.elf' failed
make[1]: *** [usbdfu.elf] Error 1
Makefile:75: recipe for target 'src/bootloader' failed
make: *** [src/bootloader] Error 2
The command '/bin/sh -c make' returned a non-zero code: 2
```

Any idea how to fix it?

--- a/src/bootloader/lctech-f103.ld
+++ b/src/bootloader/lctech-f103.ld
@@ -28,5 +28,4 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f1.ld
-
+INCLUDE cortex-m-generic.ld

--- a/src/lctech-f103.ld
+++ b/src/lctech-f103.ld
@@ -28,5 +28,4 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f1.ld
-
+INCLUDE cortex-m-generic.ld

--- a/src/stm32f4-discovery.ld
+++ b/src/stm32f4-discovery.ld
@@ -28,5 +28,5 @@ MEMORY
 }
 
 /* Include the common ld script. */
-INCLUDE libopencm3_stm32f4.ld
+INCLUDE cortex-m-generic.ld
 

